cpp/oneapi/dal/backend/primitives/lapack/solve_dpc.cpp

```diff
@@ -59,6 +61,167 @@ inline sycl::event beta_copy_transform(sycl::queue& queue,
+        return mkl::blas::column_major::gemv(queue,
+                                             mkl::transpose::nontrans,
+                                             dim_A,
+                                             num_taken,
+                                             Float(1),
+                                             Q_mutable + eigenvectors_offset,
+                                             dim_A,
+                                             ev_mutable,
+                                             1,
+                                             Float(0),
+                                             b.get_mutable_data(),
+                                             1,
+                                             { gemv_right_event });
+    }
+
+    else {
+        sycl::event gemm_right_event =
+            mkl::blas::column_major::gemm(queue,
+                                          mkl::transpose::trans,
+                                          mkl::transpose::nontrans,
+                                          num_taken,
+                                          nrhs,
+                                          dim_A,
+                                          Float(1),
+                                          Q_mutable + eigenvectors_offset,
+                                          dim_A,
+                                          b.get_data(),
+                                          dim_A,
+                                          Float(0),
+                                          ev_mutable,
+                                          num_taken,
+                                          { inv_sqrt_eigenvectors_event, event_b });
+        return mkl::blas::column_major::gemm(queue,
+                                             mkl::transpose::nontrans,
+                                             mkl::transpose::nontrans,
+                                             dim_A,
+                                             nrhs,
+                                             num_taken,
+                                             Float(1),
+                                             Q_mutable + eigenvectors_offset,
+                                             dim_A,
+                                             ev_mutable,
+                                             num_taken,
+                                             Float(0),
+                                             b.get_mutable_data(),
+                                             dim_A,
+                                             { gemm_right_event });
+    }
+}
+
+/* Returns the minimum value among entries in the diagonal of a square matrix */
+template <typename Float>
+Float diagonal_minimum(sycl::queue& queue,
+                       const Float* Matrix,
+                       const std::int64_t dim_matrix,
+                       sycl::event& event_Matrix) {
```

Comment:
Can it be potentially reused in scenario with more than 1 event? Should it be extended with const bk::event_vector& deps

Reply:
It should be theoretically possible, but the function right now is only used in this file.